.github/workflows/track-performance.yml

```
@@ -0,0 +1,205 @@
+            
+            # Wait for GitHub API to register the workflow run (retry up to 3min)
+            for i in {1..18}; do
+              sleep 10
+              RUN_ID=$(gh run list \
+                --repo ava-labs/avalanchego \
+                --workflow "Firewood Reexecution Benchmark" \
+                --limit 5 \
+                --json databaseId,createdAt \
+                --jq '[.[] | select(.createdAt | fromdateiso8601 > (now - 180))] | .[0].databaseId')
+              
+              if [ -n "$RUN_ID" ] && [ "$RUN_ID" != "null" ]; then
+                break
+              fi
+            done
+            
+            if [ -z "$RUN_ID" ] || [ "$RUN_ID" = "null" ]; then
+              echo "Could not find triggered workflow run after 3min"
+              exit 1
+            fi
+          fi
+          
+          echo "run_id=$RUN_ID" >> "$GITHUB_OUTPUT"
+          echo "run_url=https://github.com/ava-labs/avalanchego/actions/runs/$RUN_ID" >> "$GITHUB_OUTPUT"
+        env:
+          GH_TOKEN: ${{ secrets.FIREWOOD_AVALANCHEGO_GITHUB_TOKEN }}
+
+      - name: Wait for benchmark completion
+        shell: nix develop ./ffi --command bash {0}
+        run: just wait-benchmark "${{ steps.trigger.outputs.run_id }}"
+        env:
+          GH_TOKEN: ${{ secrets.FIREWOOD_AVALANCHEGO_GITHUB_TOKEN }}
+        timeout-minutes: 60
+
+      - name: Download benchmark results
+        id: download
+        shell: nix develop ./ffi --command bash {0}
+        run: |
+          just download-benchmark-results "${{ steps.trigger.outputs.run_id }}"
+          
+          # Determine target dashboard
+          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
+            echo "data-dir=bench" >> "$GITHUB_OUTPUT"
+          else
+            SAFE_NAME=$(echo "${{ github.ref_name }}" | tr '/' '-')
+            echo "data-dir=dev/bench/$SAFE_NAME" >> "$GITHUB_OUTPUT"
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.FIREWOOD_AVALANCHEGO_GITHUB_TOKEN }}
+
+      - name: Store benchmark results
+        id: store
+        continue-on-error: true
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: C-Chain Reexecution Performance
+          tool: 'go'
+          output-file-path: ./results/benchmark-output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+          gh-pages-branch: benchmark-data
+          benchmark-data-dir-path: ${{ steps.download.outputs.data-dir }}
+          fail-on-alert: false
+          comment-on-alert: false
+
+      - name: Summary
+        run: |
+          if [ "${{ steps.store.outcome }}" == "failure" ]; then
+            echo "::warning::Benchmark storage failed - results were not saved to GitHub Pages"
+          fi
+          echo "## Firewood Performance Benchmark Results" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "**Configuration:**" >> $GITHUB_STEP_SUMMARY
+          
```

Group all these together (probably with the ones below too):
```suggestion
          { 
              echo "## Firewood Performance Benchmark Results"
              echo
              echo "**Configuration:**"
              # ...
        } > $GITHUB_STEP_SUMMARY
```

It's a little particular about spacing though. There must be a space after the opening { and a newline or a semicolon ; before the closing }.